src/pages/order-service/OrderServiceControl.tsx

```diff
@@ -0,0 +1,398 @@
+  brand: {
+    name: string;
+  };
+  type: string;
+  id: string;
+  model: string;
+  unit: {
+    name: string;
+    localization: string;
+  };
+}
+
+export interface OrderServiceData {
+  id: string
+  date: Date
+  description?: string
+  authorId: string
+  receiverName: string
+  sender?: string
+  equipmentSnapshot: any
+  senderFunctionalNumber: string
+  createdAt: Date
+  updatedAt: Date
+  equipment: Equipment
+  history: History
+  receiverFunctionalNumber: string
+  technicians?: string[]
+  status: string
+  receiverDate?: Date
+}
+
+type FilterValues = {
+  type?: ISelectOption;
+  brand?: string;
+  DateOS?: string;
+  unit?: ISelectOption;
+  situation?: ISelectOption;
+  search: string;
+};
+
+function OrderServiceTable() {
+    const [orderServices, setOrderServices] = useState<OrderServiceData[]>([]);
+    const [nextOrderServices, setNextOrderServices] = useState<OrderServiceData[]>([]);
+  
+    const [selectedOrderServiceToEdit, setSelectedOrderServiceToEdit] =
+    useState<OrderServiceData>();
+    const [refreshRequest, setRefreshRequest] = useState<boolean>(false);
+    // const [workstations, setWorkstations] = useState<ISelectOption[]>();
+
+    const [currentPage, setCurrentPage] = useState(1);
+    const [offset, setOffset] = useState(0);
+    const limit = 10;
+    const [filter, setFilter] = useState<string>('');
+    const [search, setSearch] = useState<string>('');
+
+    const {
+      control,
+      watch,
+      register,
+      formState: { errors },
+    } = useForm<FilterValues>({ mode: 'onChange' });
+  
+    const watchFilter = watch();
+  
+    const handleFilterChange = () => {
+      const { type, DateOS, situation } = watchFilter;
+  
+      let formattedDate;
+      if (
+        DateOS !== null &&
+        DateOS !== '' &&
+        DateOS
+      ) {
+        formattedDate = new Date(DateOS).toLocaleDateString('en-us');
+        console.log(formattedDate);
+      }
+
+      const dataFormatted = {
+        type: type?.value,
+        updatedAt: formattedDate,
+        situation: situation?.value,
+        // unit: unit?.value,
+        search,
+      };
+  
+      const filteredDataFormatted = [
+        ...Object.entries(dataFormatted).filter(
+          (field) => field[1] !== undefined && field[1] !== ''
+        ),
+      ];
+  
+      const query = `${filteredDataFormatted
+        .map((field) => `${field[0]}=${field[1]}`)
+        .join('&')}`;
+      setFilter(query);
+  };
+
+
+    /*const formattedWorkstations = (data: Workstation[]): ISelectOption[] => {
+      return data?.map((item: Workstation) => {
+        return { label: item.name, value: item.name };
+      });
+    };*/
+
+    /* const getWorkstations = async () => {
+      apiSchedula
+        .get<Workstation[]>('workstations')
+        .then((response) => {
+        setWorkstations(formattedWorkstations(response.data));
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };*/
```

Comment:
Pode retirar esse código comentado, se for ser utilizado na filtragem basta verificar como foi feito em outra tela de listagem.